OpenRA.Game/Traits/Player/FrozenActorLayer.cs

```diff
@@ -119,6 +125,9 @@ public IEnumerable<IRenderable> Render(WorldRenderer wr)
 				}
 			}
 
+			if (Shrouded)
+				return NoRenderables;
```

Comment:
`FrozenActorLayer.Render` currently deals with filtering out invisible actors, I think you should move this check there instead.